### PR TITLE
spotify, renpy: fix wrong ffmpeg references

### LINKED ISSUE #216330: Build failure: spotify (failed to create symbolic link)
### Steps To Reproduce

1. build *spotify*

### Build log

```console
$ NIXPKGS_ALLOW_UNFREE=1 nix-build -A spotify
this derivation will be built:
  /nix/store/aqnj1f84mzsghzdbigj38740jmhv34s0-spotify-1.1.84.716.gc5f8b819.drv
building '/nix/store/aqnj1f84mzsghzdbigj38740jmhv34s0-spotify-1.1.84.716.gc5f8b819.drv'...
unpacking sources
Parallel unsquashfs: Using 16 processors
32 inodes (2300 blocks) to write
created 31 files
created 11 directories
created 1 symlink
created 0 devices
created 0 fifos
created 0 sockets
created 0 hardlinks
patching sources
configuring
no configure script, doing nothing
building
no Makefile or custom buildPhase, doing nothing
installing
ln: failed to create symbolic link './spotify': File exists
error: builder for '/nix/store/aqnj1f84mzsghzdbigj38740jmhv34s0-spotify-1.1.84.716.gc5f8b819.drv' failed with exit code 1;
       last 10 log lines:
       > created 0 fifos
       > created 0 sockets
       > created 0 hardlinks
       > patching sources
       > configuring
       > no configure script, doing nothing
       > building
       > no Makefile or custom buildPhase, doing nothing
       > installing
       > ln: failed to create symbolic link './spotify': File exists
       For full logs, run 'nix log /nix/store/aqnj1f84mzsghzdbigj38740jmhv34s0-spotify-1.1.84.716.gc5f8b819.drv'.
```

### Additional context

It was building with a few commits behind, but there were no updates to the `spotify` package, so I'm not sure why it fails now.

### Notify maintainers

@eelco @ftrvxmtrx @sheenobu @mudri @timokau @ma27 

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.9, NixOS, 23.05 (Stoat)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.13.2`
 - channels(root): `"nixos-22.05pre343321.78cd22c1b86"`
 - channels(gvolpe): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

### PR DESCRIPTION
**Pull request by GaetanLepage**
###### Description of changes

The ffmpeg library derivation output is `ffmpeg.lib` and no more `ffmpeg.out`.
This PR updates the remaining references to `ffmpeg.out`.
Fixes: https://github.com/NixOS/nixpkgs/issues/216330

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

renpy: @ShadowRZ
spotify: @ftrvxmtrx @sheenobu @laMudri @timokau @Ma27

### CODE REVIEW

**Comment by mweinelt** [comment]
@ofborg build renpy
@ofborg build spotify

**Comment by mweinelt** [comment]
Can you put the rationale into the commit message as well? That way we don't have to rely on external sources, like GitHub, to have context.

File: pkgs/applications/audio/spotify/default.nix [hunk 1]
```diff
@@ -134,8 +134,8 @@ stdenv.mkDerivation {
       ln -s ${nspr.out}/lib/libnspr4.so $libdir/libnspr4.so
       ln -s ${nspr.out}/lib/libplc4.so $libdir/libplc4.so
 
-      ln -s ${ffmpeg.out}/lib/libavcodec.so* $libdir
-      ln -s ${ffmpeg.out}/lib/libavformat.so* $libdir
+      ln -s ${ffmpeg.lib}/lib/libavcodec.so* $libdir
+      ln -s ${ffmpeg.lib}/lib/libavformat.so* $libdir
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/spotify/default.nix:138]: FYI if ``lib.getLib`` would have been used here, this could have been prevented.

**Review by ShadowRZ — COMMENTED** [review]
LGTM
Sorry to be this late 😅

**Pull request merged**